src/hooks/useReactions.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+import { useFirebase } from "react-redux-firebase";
+import { REACTION_TIMEOUT } from "settings";
+import { Reaction } from "utils/reactions";
+
+export const useReactions = (venueId?: string) => {
+  const [reactions, setReactions] = useState<Reaction[]>([]);
+  const firebase = useFirebase();
+
+  useEffect(() => {
+    if (!venueId) return;
+
+    firebase
+      .firestore()
+      .collection(`experiences/${venueId}/reactions`)
+      .where("created_at", ">", Date.now())
+      .onSnapshot((snapshot) => {
+        snapshot.docChanges().forEach((change) => {
+          if (change.type === "added") {
+            const newReaction = change.doc.data() as Reaction;
```

Comment:
Is it possible to somehow omit the type assignment here? Not sure how to do it, just asking.

Reply:
Yes, it would be great! These casts hide potential `undefined` values that bypass TypeScript and crash at runtime.

@0xdevalias has suggested a couple of ORMs we could use and I'd be very happy to see us move in that direction:

* https://github.com/wovalle/fireorm
* https://fireadmin.io/
* https://github.com/prescottprue/fireadmin

Reply:
Something along these lines is being worked on in https://github.com/sparkletown/sparkle/pull/744 and this will prevent crashes in all sorts of places in the code. Exciting to see the code get more stable.